src/RpcClient.ts

```diff
@@ -201,12 +207,21 @@ export class PostMessageRpcClient extends RpcClient {
                     console.error(`postMessage failed: ${e}`);
                 }
 
-                connectTimer = window.setTimeout(tryToConnect, 100);
+                window.setTimeout(tryToConnect, 100);
             };
 
-            connectTimer = window.setTimeout(tryToConnect, 100);
+            window.setTimeout(tryToConnect, 100);
         });
     }
+
+    private _cleanUp() {
+        this._waitingRequests.clear();
+        this._responseHandlers.clear();
+        if (this._target && this._target.closed) {
+            this._target = null;
+            window.removeEventListener('message', this._receiveListener);
+        }
+    }
```

Comment:
Should the clean-up also include a reset of the `this._connected` status to `false`. Because I think it is not set back to `false` anywhere else automatically (only in the public close() function, but that is only called from outside)?

Also, are `_waitingRequests` already rejected when this function runs? Or do they get unresolved deleted?

Reply:
The `_connected` flag already works correctly the way it is currently implemented.
When calling `close` it is set to `false`; when the popup is closed, it doesn't matter anymore as all calls check whether the popup is still open.
Setting `_connected` to false here could be done, however then the check for `_connected` in `_call` also has to be changed as when the popup was closed, there is no way to recover from that state by calling `init` again.
For completeness, I can do that though.

When the popup gets closed, all pending requests get rejected.
The other case that invokes `_cleanUp` is calling `close`.
In that case, the requests stay pending indeed and should get rejected.
I'll change that.